services-custom/s3-transfer-manager/src/it/java/software/amazon/awssdk/transfer/s3/S3TransferManagerUploadDirectoryUnicodeIntegrationTest.java

```diff
@@ -0,0 +1,169 @@
+import software.amazon.awssdk.services.s3.model.S3Object;
+import software.amazon.awssdk.testutils.FileUtils;
+import software.amazon.awssdk.utils.Logger;
+
+/**
+ * Tests the behavior of traversing local directories with special Unicode characters in their path name. These characters have
+ * known to be problematic when using Java's old File API or with Windows (which uses UTF-16 for file-name encoding).
+ */
+@RunWith(Parameterized.class)
+public class S3TransferManagerUploadDirectoryUnicodeIntegrationTest extends S3IntegrationTestBase {
+    private static final Logger log = Logger.loggerFor(S3TransferManagerUploadDirectoryUnicodeIntegrationTest.class);
+    private static final String TEST_BUCKET = temporaryBucketName(S3TransferManagerUploadDirectoryUnicodeIntegrationTest.class);
+
+    private static S3TransferManager tm;
+    private static S3Client s3Client;
+
+    private final String directoryPrefix;
+    private Path testDirectory;
+
+    @Parameterized.Parameters(name = "Directory Prefix: {0}")
+    public static Collection<Object> data() {
+        return Arrays.asList(
+            /* ASCII, 1-byte UTF-8 */
+            "E",
+            /* ASCII, 2-byte UTF-8 */
+            "É",
+            /* Non-ASCII, 2-byte UTF-8 */
+            "Ũ",
+            /* Non-ASCII, 3-byte UTF-8 */
+            "स",
+            /* Non-ASCII, 4-byte UTF-8 */
+            "\uD808\uDC8C"
+        );
+    }
+
+    public S3TransferManagerUploadDirectoryUnicodeIntegrationTest(String directoryPrefix) {
+        this.directoryPrefix = directoryPrefix;
+        log.info(() -> "Testing directory prefix: " + directoryPrefix);
+        log.info(() -> "UTF-8 bytes: " + Hex.encodeHexString(directoryPrefix.getBytes(StandardCharsets.UTF_8)));
+        log.info(() -> "UTF-16 bytes: " + Hex.encodeHexString(directoryPrefix.getBytes(StandardCharsets.UTF_16)));
+    }
+
+    @BeforeClass
+    public static void setup() throws Exception {
+        S3IntegrationTestBase.setUp();
+        tm = S3TransferManager.builder()
+                              .s3ClientConfiguration(b -> b.credentialsProvider(CREDENTIALS_PROVIDER_CHAIN)
+                                                           .region(DEFAULT_REGION)
+                                                           .maxConcurrency(100))
+                              .build();
+        s3Client = S3Client.builder()
+                           .credentialsProvider(CREDENTIALS_PROVIDER_CHAIN).region(DEFAULT_REGION)
+                           .build();
+    }
+
+    @Before
+    public void testSetup() throws Exception {
+        createBucket(TEST_BUCKET);
+        testDirectory = createLocalTestDirectory();
+    }
+
+    @After
+    public void testTeardown() {
+        try {
+            FileUtils.cleanUpTestDirectory(testDirectory);
+        } catch (Exception exception) {
+            log.warn(() -> "Failed to clean up test directory " + testDirectory, exception);
```

Comment:
Should't this be an error log ?

Reply:
IMO it doesn't really matter, the log level for all tests is `warn`.